omnigibson/envs/env_base.py

```diff
@@ -533,6 +531,11 @@ def _populate_info(self, info):
 
     def _pre_step(self, action):
         """Apply the pre-sim-step part of an environment step, i.e. apply the robot actions."""
+
+        # record the start time
+        # record the start time of the simulation step in the beginning of the step
+        self._cur_sim_start_ts = time.perf_counter()
```

Comment:
should we move these logic into the metrics themselves? e.g. the env class just call a bunch of pre_step for each of the metrics, something like this:
```
for metric in self.task.metrics:
    metric.pre_step()
```